test/index.test.ts

```diff
@@ -1,21 +1,77 @@
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 import { expect } from "chai";
-import { Contract } from "ethers";
-import { FormatTypes, Interface, parseEther } from "ethers/lib/utils";
+import { BigNumber, Contract } from "ethers";
+import {
+  formatEther,
+  FormatTypes,
+  Interface,
+  parseEther,
+} from "ethers/lib/utils";
 import { ethers } from "hardhat";
 
 import { ToucanClient } from "../dist";
-import { IToucanCarbonOffsets } from "../dist/typechain";
-import { poolTokenABI, swapperABI, tco2ABI } from "../dist/utils/ABIs";
+import { PoolSymbol } from "../dist/types";
+import { poolTokenABI, swapperABI } from "../dist/utils/ABIs";
 import addresses from "../dist/utils/addresses";
 
-describe("Testing Toucan-SDK", function () {
+const ONE_ETHER = parseEther("1.0");
+
+describe("Testing Toucan-SDK contract interactions", function () {
   let addr1: SignerWithAddress;
   let addr2: SignerWithAddress;
   let toucan: ToucanClient;
   let swapper: Contract;
-  let scoredTCO2sBCT: string[];
-  let scoredTCO2sNCT: string[];
+
+  /**
+   *
+   * @param poolSymbol Pool symbol to get scored TCO2s for
+   * @returns an array of addresses representing TCO2s in the given pool that have a balance > 0.0
+   */
+  const getFilteredScoredTCO2s = async (
+    poolSymbol: PoolSymbol
+  ): Promise<string[]> => {
+    const scoredTCO2s = await toucan.getScoredTCO2s(poolSymbol);
+    const pool = toucan.getPoolContract(poolSymbol);
+    return await asyncFilter(scoredTCO2s, async (tco2Address: string) => {
+      const tokenBalance: BigNumber = await pool["tokenBalances"](tco2Address);
+      return tokenBalance.gt(parseEther("0.0"));
+    });
+  };
+
+  const asyncFilter = async (
+    arr: string[],
+    predicate: (tco2Address: string) => Promise<boolean>
+  ) =>
+    Promise.all(arr.map(predicate)).then((results) =>
+      arr.filter((_el, index) => results[index])
+    );
+
+  /**
+   *
+   * @description checks whether you can redeem 1 TCO2 and returns the amount of redeemable TCO2
+   * @dev this exists because when redeeming you might end up attempting to redeem more than the balance of the first
+   * scored TCO2 if using a hardcoded redeem amount
+   * @param poolSymbol pool to get dynamic redeem amount for
+   * @param tco2Quality quality of the TCO2 to redeem
+   * @returns BigNumber representing the amount of TCO2 you can redeem safely
+   */
+  const getDynamicRedeemAmount = async (
+    poolSymbol: PoolSymbol,
+    tco2Quality: "low" | "high"
```

Comment:
I am not sure we want to introduce this type of API here - not against it but I know our carbon experts are not fans of the concept of quality cc @0xandi 

@marceljay @aspiers thoughts?

Reply:
Actually this is just a test, please disregard :)